src/PostgREST/DbStructure.hs

```diff
@@ -195,22 +194,46 @@ accessibleProcs = H.Statement (toS sql) (param HE.text) decodeProcs True
 
 procsSqlQuery :: SqlQuery
 procsSqlQuery = [q|
+  WITH RECURSIVE
+  rec_types AS (
+    SELECT
+      oid,
+      typbasetype,
+      COALESCE(NULLIF(typbasetype, 0), oid) AS base
+    FROM pg_type
```

Comment:
Did you checked this change against the `big schema` and see if there is a downgrade in perf?

From the looks of it, it doesn't seem so. But double checking would be good.

Reply:
Just ran the new query on the big schema and it's about `7 ms` slower than the previous query. So the perf drop is negligible, it should be fine.

Reply:
You guessed right:

```
query  | plan | execution | rows 
master | 0.37 | 6.3       | 3152
#1654  | 0.56 | 7.9       | 3152
```

It's a tiny bit slower. That makes sense, because the query is a bit more complex. But the 1.5ms here is nothing against the source columns query.

Reply:
Cool! How about adding a comment that this part is related to solving domains of domains:
```suggestion
 -- Recursively get the base types of domains
  WITH RECURSIVE
  rec_types AS (
    SELECT
      oid,
      typbasetype,
      COALESCE(NULLIF(typbasetype, 0), oid) AS base
    FROM pg_type
```